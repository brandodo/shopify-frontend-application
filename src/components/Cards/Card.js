import Moment from "react-moment";
import { SpeechButton } from "./SpeechButton";

export const Card = ({ data }) => {
  const { id, prompt, choices, created, engine, avatar } = data;

  let curator;

  switch (engine) {
    case "text-davinci-002":
      curator = "DaVinci";
      break;

    case "text-curie-001":
      curator = "Madam Curie";
      break;

    case "text-babbage-001":
      curator = "Count Babbage";
      break;

    case "text-ada-001":
      curator = "Apparition Ada";
      break;

    default:
      break;
  }

  return (
    <div id={id} data-testid="card" className="cards__card">
      <div className="cards__card-container">
        <div>
          <h4 className="cards__card-header">
            Prompt: <span>{prompt}</span>
          </h4>
          <p className="cards__card-text">
            <span>Curator: </span>
            {`${curator} (${engine})`}
          </p>
        </div>
        <img className="cards__card-avatar" src={avatar} />
      </div>
      <SpeechButton text={choices[0].text} engine={engine} />
      <p className="cards__card-text">{choices[0].text}</p>
      <div className="cards__card-date">
        <p className="cards__card-text">Created on: </p>
        <Moment
          className="cards__card-text"
          unix
          format="MMM DD, YYYY HH:mm:ss"
        >
          {created}
        </Moment>
      </div>
    </div>
  );
};
