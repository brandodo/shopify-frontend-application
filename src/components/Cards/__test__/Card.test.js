import React from "react";
import ReactDOM from "react-dom";
import { Card } from "../Card";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");

  const data = {
    id: "12312312",
    prompt: "hello",
    choices: ["what's up", "ayo", "wow"],
    created: 534534543345,
    engine: "text-curie-001",
    avatar: "../../assets/images/curie.png",
  };

  ReactDOM.render(<Card data={data} />, div);
});

it("renders correctly", () => {
  const data = {
    id: "wow-123123",
    prompt: "hello",
    choices: ["fdafafassf", "ayo", "wow"],
    created: 534534543345,
    engine: "text-davinci-002",
    avatar: "../../assets/images/davinci.png",
  };

  const { getByTestId } = render(<Card data={data} />);
  expect(getByTestId("card")).toBeTruthy();
});

it("renders correctly", () => {
  const data = {
    id: "opl123123",
    prompt: "roooo",
    choices: ["weaesa", "aadsasdasdyo", "dasasdadsasd", "dasadsasd"],
    created: 534534543345,
    engine: "text-babbage-001",
    avatar: "../../assets/images/babbage.png",
  };

  const { getByTestId } = render(<Card data={data} />);
  expect(getByTestId("card")).toBeTruthy();
});

it("matches snapshot", () => {
  const data = {
    id: "weeeee-123123",
    prompt: "roooo",
    choices: ["weaesa", "aadsasdasdyo", "dasasdadsasd", "dasadsasd"],
    created: 534534543345,
    engine: "text-ada-001",
    avatar: "../../assets/images/ada.png",
  };

  const tree = renderer.create(<Card data={data} />).toJSON();

  expect(tree).toMatchSnapshot();
});
