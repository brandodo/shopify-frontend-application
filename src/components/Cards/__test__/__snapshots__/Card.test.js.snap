// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="cards__card"
  data-testid="card"
  id="weeeee-123123"
>
  <div
    className="cards__card-container"
  >
    <div>
      <h4
        className="cards__card-header"
      >
        Prompt: 
        <span>
          roooo
        </span>
      </h4>
      <p
        className="cards__card-text"
      >
        <span>
          Curator: 
        </span>
        Apparition Ada (text-ada-001)
      </p>
    </div>
    <img
      className="cards__card-avatar"
      src="../../assets/images/ada.png"
    />
  </div>
  <button
    className="cards__button"
    onClick={[Function]}
  >
    Read it to me!
  </button>
  <p
    className="cards__card-text"
  />
  <div
    className="cards__card-date"
  >
    <p
      className="cards__card-text"
    >
      Created on: 
    </p>
    <time
      className="cards__card-text"
      dateTime={"+018908-09-22T09:35:45.000Z"}
    >
      Sep 22, 18908 05:35:45
    </time>
  </div>
</div>
`;
